### PR TITLE
Feature/editor refactoring

### PR DESCRIPTION
**Pull request by ResearchDaniel**
Sorry for this huge PR. Will try to split things into smaller pieces.

frontend:
- Wrap image index + abstraction level into for consistent handling
- Make reading mode info top level
- Components for view modes
- Separate editing into an Editor
- Add dependency: flowbite-svelte

backend:
- read routes from separate file through Blueprints

New look:
image

### CODE REVIEW

File: frontend/src/lib/types.ts [hunk 1]
```diff
@@ -24,22 +65,50 @@ export interface SelectedImages {
...
+// 				return this.bookSelectedId;
+// 			case AbstractionLevel.CHAPTER:
+// 				return this.chapters[imageIndex.level.chapterId].chapterSelectedId;
+// 			case AbstractionLevel.PARAGRAPH:
+// 				return this.chapters[imageIndex.level.chapterId].paragraphSelectedIds[
+// 					imageIndex.level.paragraphId
+// 				];
+// 		}
+// 	}
+// }
```
Comment (Sparkier) [inline frontend/src/lib/types.ts:106]: Can this be deleted?

**Review by Sparkier — APPROVED** [review]

**Comment by ResearchDaniel** [comment]
@Sparkier should I merge?

**Comment by Sparkier** [comment]
fine to merge from my side, only thing is that we already have some components (e.g. button) that you're now adding different versions for with flowbite. Maybe we should commit to only one of the two?

**Comment by ResearchDaniel** [comment]
> fine to merge from my side, only thing is that we already have some components (e.g. button) that you're now adding different versions for with flowbite. Maybe we should commit to only one of the two?

Will do that in future PR

**Pull request merged**